docs/DeveloperGuide.adoc

```diff
@@ -354,16 +361,236 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+.The current queue
+image::Figure 3.2.1.1.PNG[width="800"]
+
+The user now decides to edit the remarks of the first person in the list.
+
+`RemarkCommand` will search for the first person in the list and store it as the person to be edited.
+`RemarkCommand` will then create a new patient with the same attributes as the patient to be edited, except the remarks attribute
+which will be based on the user input. This new patient will be stored as the edited patient as shown in the figure below.
+
+.The patient before and after editing
+image::Figure 3.2.1.2.PNG[width="800"]
+
+`RemarkCommand` will finally call the for the utility method updatePerson under `Model`, to replace the patient to be edited with the edited patient,
+the new list will be as shown in the figure below.
+
+.The queue with updated details
+image::Figure 3.2.1.3.PNG[width="800"]
+
+==== Design Considerations
+
+The following is an exploration of possible alternative implementations, the various aspects of
+each implementation and the pros and cons of each implementation:
+
+===== Aspect: Implementation of `RemarkCommand`
+
+* **Alternative 1 (current choice):** Create a duplicate `Patient` and utilise updatePerson to replace the `Patient`
+** Pros: Will be easy to implement due to existing utility methods.
+** Cons: May have performance issues in terms of memory usage.
+* **Alternative 2:** Create a public method in `Remark` for updating the remark which will be called by `Patient`
+** Pros: Will use less memory (no need to create new patient object).
+** Cons: Will require public access to `Remark` stored in `Patient` (possible security concerns).
+
+===== Aspect: Data structure to support the remark command
+
+* **Alternative 1 (current choice):** Create a `Remark` that is stored in `Patient`
+** Pros: Will be easy to understand for new developers as it follows existing structures.
+** Cons: May have performance issues in terms of memory usage.
+* **Alternative 2:** Create a String that is stored in `Patient`
+** Pros: Will use less memory (no need to create new remark object).
+** Cons: Will require a method in Patient to update and verify validity of remark. Violates Single Responsibility Principle and Separation of Concerns as `Patient` now needs to do two different things.
+// end::remark[]
+
+// tag::morefields[]
+=== Implement NRIC, DOB and Blood Type detail fields
+==== Current Implementation
+
+The additional fields added is managed by the `AddCommand` and the `EditCommand`.
+They are included as fields to be filled in when adding a new `Patient` or when editing an existing `Patient`.
+
+Both `AddCommand` and `EditCommand` have been modified to accept new inputs corresponding with the new fields implemented.
+
+Suppose that the user has just launched the application. The current list is shown in Figure 22 below.
+
+.The current queue
+image::Figure3-3-1-1.png[width="800"]
```

Comment:
Check the aspect ratios of the images. Try to make them not too big. Sequence Diagrams are effective in visualization of the functioning of features. You may consider adding them.